pkg/corerp/api/v20231001preview/environment_conversion.go

```diff
@@ -408,35 +420,118 @@ func fromRecipePropertiesClassificationDatamodel(e datamodel.EnvironmentRecipePr
 	return nil
 }
 
-func toRecipeConfigTerraformProvidersDatamodel(config *RecipeConfigProperties) map[string][]datamodel.ProviderConfigProperties {
+func toRecipeConfigTerraformProvidersDatamodel(config *RecipeConfigProperties) (map[string][]datamodel.ProviderConfigProperties, error) {
 	if config.Terraform == nil || config.Terraform.Providers == nil {
-		return nil
+		return nil, nil
 	}
 
 	dm := map[string][]datamodel.ProviderConfigProperties{}
 	for k, v := range config.Terraform.Providers {
 		dm[k] = []datamodel.ProviderConfigProperties{}
 
 		for _, providerAdditionalProperties := range v {
+			var addnlProperties map[string]any
```

Comment:
we don't really abbreviate `additional` to `addnl` anywhere else in the code base

```suggestion
			var additionalProperties map[string]any
```